### ISSUE TITLE
Make sure CoT gap is measured consistently in loop

### ISSUE THREAD

**Issue by puria-radmard** [puria-radmard/mars-steg#28]
At the end of the main training loop there is somewhere where we can do this

Use prompt data batch item (@jskaf34 can advise) to get back CoT gap during training

Make sure no gradients are passed

**Comment by jskaf34** [comment]
A small question here : so we will run the model twice on the same prompt right ? So we will double generation/inference time ?

**Comment by jskaf34** [comment]
Done : I will explain it further during meeting, but we chose with Connor to evaluate at every epoch the CoT Gap on a validation dataset (very first step, even if the validation dataset can be victim of data contamination, blank tokens can still be a great option to add in the function `batchize_conversation` win `evaluate_cot_gap_and_ratio`

What can be done in the future as well : evaluate CoT Gap on training dataset to mesure as well the overfitting of the model.
